azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventPosition.java

```diff
@@ -0,0 +1,179 @@
+     * Instead, the next event is returned.
+     * @param sequenceNumber is the sequence number of the event.
+     * @return An {@link EventPosition} object.
+     */
+    public static EventPosition fromSequenceNumber(Long sequenceNumber) {
+        return EventPosition.fromSequenceNumber(sequenceNumber, false);
+    }
+
+    /**
+     * Creates a position at the given sequence number. The specified event will not be included.
+     * Instead, the next event is returned.
+     * @param sequenceNumber    is the sequence number of the event.
+     * @param inclusiveFlag         will include the specified event when set to true; otherwise, the next event is returned.
+     * @return An {@link EventPosition} object.
+     */
+    public static EventPosition fromSequenceNumber(Long sequenceNumber, boolean inclusiveFlag) {
+        return new EventPosition(null, sequenceNumber, null, inclusiveFlag);
+    }
+
+    /**
+     * Creates a position at the given {@link Instant}.
+     * @param dateTime  is the enqueued time of the event.
+     * @return An {@link EventPosition} object.
+     */
+    public static EventPosition fromEnqueuedTime(Instant dateTime) {
+        return new EventPosition(null, null, dateTime, null);
+    }
+
+    /**
+     * This will return an EventPosition that includes the first event in the Event Hubs partition.
+     *
+     * @return An {@link EventPosition} with position set to the start of the Event Hubs stream.
+     */
+    public static EventPosition fromStartOfStream() {
+        return new EventPosition(PartitionReceiver.START_OF_STREAM, null, null, true);
+    }
+
+    /**
+     * The last event in the Event Hubs partition is not included. The returned EventPosition will
+     * future events that arrive to this EventHubs partition.
+     *
+     * @return An {@link EventPosition} with position set to the end of the Event Hubs stream.
+     */
+    public static EventPosition fromEndOfStream() {
+        return new EventPosition(PartitionReceiver.END_OF_STREAM, null, null, false);
+    }
+
+    /**
+     * Gets the offset of the event at the position. It can be null if the position is created
+     * from a sequence number or enqueued time.
+     *
+     * @return the byte offset of the event.
+     */
+    public String getOffset() {
+        return this.offset;
+    }
+
+    /**
+     * Indicates if the current event at the specified offset is included or not.
+     * It is only applicable if offset or sequence number is set. If offset or
+     * sequence number is not set, this will return null.
+     *
+     * @return the inclusive flag
+     */
+    public Boolean getInclusiveFlag() {
+        return this.inclusiveFlag;
+    }
+
+    /**
+     * Gets the sequence number of the event at the position. It can be null if the position is created
+     * from an offset or enqueued time.
+     *
+     * @return the sequence number of the event.
+     */
+    public Long getSequenceNumber() {
```

Comment:
>getSequenceNumber [](start = 16, length = 17)

Consensus after discussion with Xin & SJ:
- lets remove the public get's
- lets retain the getLatestSequenceNo, getLatestOffset on PartitionReceiver